### PR TITLE
functional survey

### PR DESCRIPTION
**Pull request by patrickkettner**

### CODE REVIEW

File: frontend/templates/views/partials/survey.j2 [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
+  [% set url = "{{ doc.url }}" %]
+
   {% do doc.styles.addCssFile('css/components/organisms/survey.css') %}
 
   <amp-state id="surveyQuestions">
     <script type="application/json">
-
-      {% do survey_questions.update({"_url": request.url })%}
-      {{ survey_questions | tojson }}
+        [% set survey = surveys|random|survey(fixedQuestions, url) %]
```
Comment (sebastianbenz) [inline frontend/templates/views/partials/survey.j2:10]: What does the random do here?

**Review by sebastianbenz — APPROVED** [review]
What was the problem? Looks good once the build passes.

File: frontend/templates/views/partials/survey.j2 [hunk 2]
```diff
@@ -1,11 +1,14 @@
 {% if survey_questions %}
+  [% set surveys = {{ survey_questions|tojson }} %]
+  [% set fixedQuestions = {{fixed_questions|tojson }} %]
+  [% set url = "{{ doc.url }}" %]
```
Comment (matthiasrohmer) [inline frontend/templates/views/partials/survey.j2:4]: Alternatively you could also do

```suggestion
  [% set url = requestPath %]
```

This is given to the SSR context here:

https://github.com/ampproject/amp.dev/blob/2e3a1fb39f46110065c01ae0f8c8e4a8e11f1965/platform/lib/templates/index.js#L44

Not too sure about the backend but you also get the format the user is on, which might help putting the answers into context:

https://github.com/ampproject/amp.dev/blob/2e3a1fb39f46110065c01ae0f8c8e4a8e11f1965/platform/lib/templates/index.js#L54

File: frontend/templates/views/partials/first_survey.j2 [hunk 3]
```diff
@@ -0,0 +1,32 @@
+{% set survey_questions = [{
```
Comment (matthiasrohmer) [inline frontend/templates/views/partials/first_survey.j2:1]: No need to localize them just yet?

File: platform/lib/templates/SurveyFilter.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
```
Comment (matthiasrohmer) [inline platform/lib/templates/SurveyFilter.js:2]: Nit: 2021

File: frontend/templates/views/partials/survey.j2 (hunk 1 above)
Comment (patrickkettner) [reply frontend/templates/views/partials/survey.j2:10]: `surveys` is an array of survey objects. `random` picks one of the available objects at runtime

File: frontend/templates/views/partials/first_survey.j2 (hunk 3 above)
Comment (patrickkettner) [reply frontend/templates/views/partials/first_survey.j2:1]: haven't sent the questions to the localization folks

Comment (patrickkettner) [reply frontend/templates/views/partials/first_survey.j2:1]: will add it seperatly

File: platform/lib/templates/SurveyFilter.js (hunk 4 above)
Comment (patrickkettner) [reply platform/lib/templates/SurveyFilter.js:2]: fixed!

**Comment by patrickkettner** [comment]
@sebastianbenz 

> What was the problem? Looks good once the build passes.

The issue I was hitting is still an issue, but it was completely orthogonal to getting it to work. @matthiasrohmer was able to set me straight, I had to pass the object from grow to SSR, and then use different tags to pull in a version at request time, rather than at deploy time.

**Pull request merged**